src/cloudflare/internal/test/d1/d1-api-test-common.js

```diff
@@ -0,0 +1,515 @@
+// Copyright (c) 2023 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+import * as assert from 'node:assert';
+
+// Recurse through nested objects/arrays looking for 'anything' and deleting that
+// key/value from both objects. Gives us a way to get expect.toMatchObject behaviour
+// with only deepEqual
+const anything = Symbol('anything');
+const deleteAnything = (expected, actual) => {
+  Object.entries(expected).forEach(([k, v]) => {
+    if (v === anything) {
+      delete actual[k];
+      delete expected[k];
+    } else if (typeof v === 'object' && typeof actual[k] === 'object') {
+      deleteAnything(expected[k], actual[k]);
+    }
+  });
+};
+
+export const itShould = async (description, ...assertions) => {
+  if (assertions.length % 2 !== 0)
+    throw new Error('itShould takes pairs of cb, expected args');
+
+  try {
+    for (let i = 0; i < assertions.length; i += 2) {
+      const cb = assertions[i];
+      const expected = assertions[i + 1];
+      const actual = await cb();
+      deleteAnything(expected, actual);
+      try {
+        assert.deepEqual(actual, expected);
+      } catch (e) {
+        console.log(actual);
+        throw e;
+      }
+    }
+  } catch (e) {
+    throw new Error(`TEST ERROR!\n❌ Failed to ${description}\n${e.message}`);
+  }
+};
+
+// Make it easy to specify only a the meta properties we're interested in
+const meta = (values) => ({
+  duration: anything,
+  served_by: anything,
+  changes: anything,
+  last_row_id: anything,
+  changed_db: anything,
+  size_after: anything,
+  rows_read: anything,
+  rows_written: anything,
+  ...values,
+});
+
+export async function testD1ApiQueriesHappyPath(DB) {
```

Comment:
This was cut-pasted as-is, without any changes apart from adding the last section of DELETE statements, so that the database is cleaned up.